### DISCUSSION TITLE
Best way to upgrade Teleport version, helm deployment

### DISCUSSION THREAD

**Discussion by urbaman** [gravitational/teleport#29953] [Q&A] [answered]
Hi,

What's the best way to upgrade the kubernetes teleport helm deployment (both cluster and kube-agent)?

- modifying the teleport version in the deployment, then rollout restart it
- upgrading through helm (risking not to remember eventual manual changes to the deployment)

Thanks,

**Comment by webvictim** [comment, marked as answer]
I would always do upgrades via Helm to make sure things stay in line. There may be small changes and bug fixes to the chart which are important to pull in.

```console
helm repo update
helm upgrade -n <namespace> <release-name> teleport/teleport-cluster -f teleport-cluster-values.yaml --version 13.3.1
helm upgrade -n <namespace> <release-name> teleport/teleport-kube-agent -f teleport-kube-agent-values.yaml --version 13.3.1
```

**Reply by urbaman** [reply]
Yes, that makes perfect sense, only I should always need to check the old values used at time of (last) revision installed to properly compare them with the new chart's version.

```
helm -n <namespace> get values <release-name> > <chart>-values.yaml
```

And eventually upgrade with `--reuse-values -f <chart>-values.yaml`
